### DISCUSSION TITLE
Yaml support?

### DISCUSSION THREAD

**Discussion by Cppowboy** [outlines-dev/outlines#360] [Q&A]
outlines supports json format output very well, but sometimes it is necessary to limit the model output to yaml format. Can this feature be supported in future versions?

**Comment by rlouf** [comment]
The cheeky answer is that YAML is a superset of JSON so any valid JSON is valid YAML. The answer that doesn't miss the point is that it will be supported with grammar-guided generation.

What's your use case?

**Reply by michalwols** [reply]
Main benefit of YAML is that it's less verbose so you can get away with using less tokens
